### PR TITLE
Handle degeneracy in spheroid surface area formula

### LINKED ISSUE #7472: skimage.draw.ellipsoid_stats doesn't give the result for a sphere if a=b=c
### Description:
Calculation of surface area and volume using skimage.draw.ellipsoid_stats doesn't converge to the properties of a sphere. 

## Expected
We should get the surface area of a sphere when the three major axis dimensions (a, b, c) are same 

## Result
Get a divide by zero error

### Way to reproduce:

```python
import skimage as ski

a = b = c = 10
ellipsoid = ski.draw.ellipsoid(a, b, c)
vol, surf = ski.draw.ellipsoid_stats(a, b, c)
print(f'Theoritical volume = {vol}\nTheoritical Surface Area = {surf}')```

##Error Message
Traceback (most recent call last):
  File "c:\Users\...\test.py", line 5, in <module>
    vol, surf = ski.draw.ellipsoid_stats(a, b, c)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\...\Lib\site-packages\skimage\draw\draw3d.py", line 105, in ellipsoid_stats
    m = (a ** 2 * (b ** 2 - c ** 2) /
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ZeroDivisionError: float division by zero

### Version information:

```Shell
3.12.2 | packaged by conda-forge | (main, Feb 16 2024, 20:42:31) [MSC v.1937 64 bit (AMD64)]
Windows-11-10.0.22631-SP0
scikit-image version: 0.22.0
numpy version: 1.26.4
```

### PR DESCRIPTION
**Pull request by stefanv**
This happened when a = b = c, i.e. when ellipsoid turns to sphere.

Also improves test accuracy.

Closes #7472

## Release note

For maintainers and optionally contributors, please refer to the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) on how to document this PR for the release notes.

```release-note
Fix degeneracy in `skimage.draw.ellipsoid_stats`
when all semi-axes have the same length.
```

### CODE REVIEW

**Comment by mkcor** [comment]
@stefanv thanks for this nice improvement!

I have taken the liberty of pushing to your branch because the GitHub interface wouldn't let me make suggestions on non-modified code. I have broken down my changes into several commits to separate the non-controversial ones from the docs-only ones and the code-impacting ones, so you can easily revert some if needed.

I'm suggesting pointing to the SciPy implementation (single source, right) and I used this referencing format: https://docs.ckan.org/en/ckan-2.2.3/python-coding-standards.html#referencing-other-code-objects-with-py

File: skimage/draw/draw3d.py [hunk 1]
```diff
@@ -1,20 +1,21 @@
 import numpy as np
-from scipy.special import ellipkinc as ellip_F, ellipeinc as ellip_E
+from scipy.special import elliprg
 
 
 def ellipsoid(a, b, c, spacing=(1.0, 1.0, 1.0), levelset=False):
-    """
-    Generates ellipsoid with semimajor axes aligned with grid dimensions
+    """Generate ellipsoid for given lengths of semi-axes.
```
Comment (stefanv) [inline skimage/draw/draw3d.py:6]: given semi-axis lengths

File: skimage/draw/draw3d.py [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
+
+    The respective semi-axes are aligned with grid dimensions
```
Comment (stefanv) [inline skimage/draw/draw3d.py:8]: What does aligned with grid dimensions mean? And how do you rotate this thing?

File: skimage/draw/draw3d.py [hunk 3]
```diff
@@ -62,18 +63,19 @@ def ellipsoid(a, b, c, spacing=(1.0, 1.0, 1.0), levelset=False):
 
 
 def ellipsoid_stats(a, b, c):
-    """
-    Calculates analytical surface area and volume for ellipsoid with
-    semimajor axes aligned with grid dimensions of specified `spacing`.
+    """Calculate analytical volume and surface area of an ellipsoid.
+
+    This function makes use of :py:func:`scipy.special.elliprg` to calculate
```
Comment (stefanv) [inline skimage/draw/draw3d.py:68]: I'd remove this line, it's an implementation detail.

**Review by stefanv — COMMENTED** [review]
@mkcor left some comments on the docs changes; feel free to change what you want to there, and then merge the PR

File: skimage/draw/draw3d.py (hunk 1 above)
Comment (mkcor) [reply skimage/draw/draw3d.py:6]: Agreed!

Comment (stefanv) [reply skimage/draw/draw3d.py:6]: Just note singular semi-axis, plural lengths

File: skimage/draw/draw3d.py (hunk 3 above)
Comment (mkcor) [reply skimage/draw/draw3d.py:68]: It is; you can remove it (this addition is commit 321b6011e70b5d60494205ab571fc62ce9d9c727).

But that's how I would address https://github.com/scikit-image/scikit-image/issues/7472#issuecomment-2237639870. 'Everyone knows' that the analytical formula for the volume of an ellipsoid is `4/3*pi*a*b*c`; but it's not clear/guessable a priori which formula will be used for the surface area, so this extra line helps the interested user know where to look and/or, at a glance, know what 'kind' of formula is being used (I mean even really vaguely, not in the sense of 'first kind, second kind, ...').

File: skimage/draw/draw3d.py (hunk 1 above)
Comment (mkcor) [reply skimage/draw/draw3d.py:6]: Yes, yes! :+1:

File: skimage/draw/draw3d.py (hunk 2 above)
Comment (mkcor) [reply skimage/draw/draw3d.py:8]: Legacy wording, hehe. Looking at the code lines 51-55, it's just a convention: They could/should be called first, second, and third dimensions. Basically it's just that you want the `a, b, c` order to match the `spacing` order.

File: skimage/draw/draw3d.py (hunk 3 above)
Comment (mkcor) [reply skimage/draw/draw3d.py:68]: What do you think of 3c7d1607eb17864b18133bcca1b2247078967cbc?

Comment (stefanv) [reply skimage/draw/draw3d.py:68]: In that case, I would write something like:

```
The surface of the ellipsoid is given by

$S=3VR_G\left(a^{-2},b^{-2},c^{-2}\right)$

where $R_G$ is one of the Carlson symmetric forms of the elliptical integrals, implemented as :py:func:`scipy.special.elliprg`.
```

Comment (stefanv) [reply skimage/draw/draw3d.py:68]: Maybe with a reference to the Wikipedia section: https://en.m.wikipedia.org/wiki/Ellipsoid#Surface_area

Comment (stefanv) [reply skimage/draw/draw3d.py:68]: > What do you think of [3c7d160](https://github.com/scikit-image/scikit-image/commit/3c7d1607eb17864b18133bcca1b2247078967cbc)?

LGTM!

Comment (mkcor) [reply skimage/draw/draw3d.py:68]: > In that case, I would write something like:
> 
> ```
> The surface of the ellipsoid is given by
> 
> $S=3VR_G\left(a^{-2},b^{-2},c^{-2}\right)$
> 
> where $R_G$ is one of the Carlson symmetric forms of the elliptical integrals, implemented as :py:func:`scipy.special.elliprg`.
> ```

@pamitabh as a specimen of interested user, what do you say? :wink:

**Review by mkcor — APPROVED** [review]

File: skimage/draw/draw3d.py (hunk 3 above)
Comment (pamitabh) [reply skimage/draw/draw3d.py:68]: Thanks for asking! sounds good, additional comment at #7472.

**Review by lagru — APPROVED** [review]
Looks good! :smiley: Holding off on merge since it looks like that the docstring should still be updated with the equation.

**Comment by mkcor** [comment]
When this PR is merged, we could contact the author of https://analyticphysics.com/ to let them know! I browsed the website quickly, and added 8a4c0bb32f21b31 so the citation is more complete.

File: skimage/draw/draw3d.py [hunk 4]
```diff
@@ -62,18 +63,24 @@ def ellipsoid(a, b, c, spacing=(1.0, 1.0, 1.0), levelset=False):
...
 
 def ellipsoid_stats(a, b, c):
-    """
-    Calculates analytical surface area and volume for ellipsoid with
-    semimajor axes aligned with grid dimensions of specified `spacing`.
+    """Calculate analytical volume and surface area of an ellipsoid.
+
+    The surface area of an ellipsoid is given by
+
+    .. math:: S=4\\pi b c R_G\\!\\left(1, \\frac{a^2}{b^2}, \\frac{a^2}{c^2}\\right)
```
Comment (lagru) [inline skimage/draw/draw3d.py:70]:
```suggestion
    .. math:: S=4\\pi b c R_G\\!\\left(\\frac{a^2}{b^2}, \\frac{a^2}{c^2}, 1\\right)
```

Shouldn't we follow the convention of the cited source which places the $1$ at the end of $R_G(...)$?

Comment (mkcor) [reply skimage/draw/draw3d.py:70]: I think it's fine because of the complete symmetry.

Comment (lagru) [reply skimage/draw/draw3d.py:70]: Ah, you are right, the order of the arguments doesn't matter. I'll merge then.

**Comment by lagru** [comment]
Thanks everyone!

**Pull request merged**